### PR TITLE
Enhance Tag theme for remove button

### LINKED ISSUE #7387: Tag — allow "Remove" to be default button with `size` prop rather than just a `plain` button
### Is this something you or your team is interested in contributing a pull request towards?
Yes.

### Expected Behavior

Tag "remove" button should be able to leverage a theme's button kinds rather than just being a `plain` button.

Caller should be able to define `kind` and `size` of remove button per Tag size.

### Actual Behavior

Not possible.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Enhances Tag theme to allow for caller to leverage one of their existing Button kinds as opposed to `plain` Button.

#### Where should the reviewer start?

src/js/components/Tag/Tag.js

#### What testing has been done on this PR?

Locally with HPE theme override in preview.js:

```
import React, { useState, useEffect } from 'react';
import Root from 'react-shadow';
import { Close } from 'grommet-icons';
import { StyleSheetManager } from 'styled-components';
import { hpe } from 'grommet-theme-hpe';
import isChromatic from 'chromatic/isChromatic';
import { Grommet, grommet, hacktoberfest2022, Box, Text } from '../src/js';
import { deepMerge } from '../src/js/utils';

const themeUpdate = deepMerge(hpe, {
  icon: {
    size: {
      xsmall: '14px',
    },
  },
  button: {
    size: {
      xsmall: {
        border: {
          radius: '2em',
        },
        iconOnly: {
          pad: '5px',
        },
      },
    },
  },
  tag: {
    icons: {
      remove: Close,
    },
    remove: {
      kind: 'default',
    },
    size: {
      xsmall: {
        icon: {
          size: undefined,
        },
        remove: {
          size: 'xsmall',
          margin: {
            right: 'none',
            left: '-5px', // shifting for button padding
            top: '-1px', // account for border
            bottom: '-1px', // account for border
          },
        },
      },
      small: {
        icon: {
          size: undefined,
        },
        remove: {
          size: 'xsmall',
          margin: {
            right: '2px',
            left: '-5px', // shifting for button padding
          },
        },
      },
      medium: {
        icon: {
          size: undefined,
        },
        remove: {
          size: 'small',
          margin: {
            right: 'xxsmall',
            left: '-7px', // shifting for button padding
          },
        },
      },
      large: {
        icon: {
          size: undefined,
        },
        remove: {
          size: 'medium',
          margin: {
            right: '2px',
            left: '-9px', // shifting for button padding
          },
        },
      },
      xlarge: {
        icon: {
          size: undefined,
        },
        pad: {
          vertical: '21px',
          horizontal: 'medium',
        },
        remove: {
          size: 'large',
          margin: {
            right: 'xsmall',
            left: '-13px', // shifting for button padding
          },
        },
      },
    },
  },
});

const CUSTOM_THEMED = 'Custom Themed';
const THEMES = {
  hpe: themeUpdate,
  grommet,
  hacktoberfest2022,
  base: {},
};

export const decorators = [
  (Story, context) => {
    const [rootRef, setRootRef] = useState(null);
    const [state, setState] = useState('grommet');
    const [root, setRoot] = useState('document');
    const full = context.allArgs?.full || 'min';
    const dir = context.allArgs?.dir;
    const options = context.allArgs?.options;

    useEffect(() => {
      setState(context.globals.theme);
    }, [context.globals.theme]);
    useEffect(() => {
      setRoot(context.globals.root);
    }, [context.globals.root]);

    /**
     * This demonstrates that custom themed stories are driven off the "base"
     * theme. Custom themed stories will live under a "CustomThemed" directory.
     */
    if (context.kind.split('/')[2] === CUSTOM_THEMED && state !== 'base') {
      // if we are running the story in chromatic we want the chromatic snapshot
      // to be taken in the base theme for custom theme stories
      if (isChromatic()) {
        return (
          <Grommet theme={THEMES.base}>
            <Story state={THEMES.base} />
          </Grommet>
        );
      }
      return (
        <Box align="center" pad="large">
          <Text size="large">
            {`Custom themed stories are only displayed in the
                "base" theme mode. To enable, select "base" from the
                Theme menu above.`}
          </Text>
          <div hidden>
            <Story state={THEMES[state]} />
          </div>
        </Box>
      );
    }

    if (root === 'shadow') {
      return (
        // eslint-disable-next-line react/jsx-pascal-case
        <Root.div ref={setRootRef}>
          {rootRef && (
            <StyleSheetManager target={rootRef.shadowRoot}>
              <Grommet
                theme={THEMES[state]}
                full={full}
                dir={dir}
                options={options}
                containerTarget={rootRef.shadowRoot}
              >
                <Story state={THEMES[state]} />
              </Grommet>
            </StyleSheetManager>
          )}
        </Root.div>
      );
    }

    return (
      <Grommet theme={THEMES[state]} full={full} dir={dir} options={options}>
        <Story state={THEMES[state]} />
      </Grommet>
    );
  },
];

export const parameters = {
  layout: 'fullscreen',
  options: {
    storySort: (first, second) => {
      /**
       * The story sort algorithm will only ever compare two stories
       * a single time. This means that every story will only ever be either
       * the "first" parameter OR the "second" parameter, but not both.
       * So, the checks for custom themed stories need to happen on both inputs
       * of this function.
       *
       * A return value of 1 results in sorting the "first" story AFTER the
       * "second" story.
       *
       * A return value of 0 results in sorting the "first" story BEFORE the
       * secondary story.
       */
      const isFirstCustom = first[1].kind.split('/')[2] === CUSTOM_THEMED;
      const isSecondCustom = second[1].kind.split('/')[2] === CUSTOM_THEMED;
      if (isFirstCustom) return 1;
      if (isSecondCustom) return 0;
      return first[1].kind === second[1].kind
        ? 0
        : first[1].id.localeCompare(second[1].id, undefined, { numeric: true });
    },
  },
};

export const globalTypes = {
  theme: {
    defaultValue: 'grommet',
    toolbar: {
      title: 'Theme',
      items: [
        { title: 'base', value: 'base' },
        { title: 'grommet', value: 'grommet' },
        { title: 'hpe', value: 'hpe' },
        { title: 'hacktoberfest2022', value: 'hacktoberfest2022' },
      ],
    },
  },
  root: {
    defaultValue: 'document',
    toolbar: {
      title: 'Root',
      items: ['document', 'shadow'],
    },
  },
};

```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

Closes #7387 

#### Screenshots (if appropriate)

https://github.com/user-attachments/assets/8e071fe2-4b9f-4507-a47c-80d2d5608bb2

#### Do the grommet docs need to be updated?
Yes.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

**Review by halocline — COMMENTED** [review]
So to confirm the theme property definitions (as well as what we will add to Grommet docs):

- `tag.remove` - The style of the remove button.
- `tag.size.[size].remove` - The style of the remove button.

**Object** 
```
{
    kind: 'default',
    size: 'small',
    margin: '1px'
}
```
where `kind` can be: 
a reference to any button defined in button's theme provided `button.default` is defined
 
where `size` can be:
"small"
"medium"
"large"
any size defined in `button.size`

where `margin` can be:
"any CSS size"

File: src/js/components/Tag/__tests__/Tag-test.tsx [hunk 1]
```diff
@@ -121,4 +121,42 @@ describe('Tag', () => {
     expect(container).toMatchSnapshot();
     expect(screen.getByLabelText('FormClose')).toBeDefined();
   });
+
+  test('renders custom remove button kind and size', () => {
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/Tag/__tests__/Tag-test.tsx:125]: Yes good call, I'll add margin.

[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/Tag/__tests__/Tag-test.tsx:125]: Done.

File: src/js/themes/base.js [hunk 2]
```diff
@@ -1869,18 +1869,22 @@ export const generate = (baseSpacing = 24, scale = 6) => {
         xsmall: {
           pad: { horizontal: `${baseSpacing / 3}px`, vertical: 'xxsmall' },
           icon: { size: `${baseSpacing * 0.75}px` },
+          // remove: {},
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/themes/base.js]: Done.

**Review by halocline — APPROVED** [review]

**Pull request merged**